nixos/modules/programs/thefuck.nix

```diff
@@ -24,8 +29,9 @@ in
 
     config = mkIf cfg.enable {
       environment.systemPackages = with pkgs; [ thefuck ];
-      environment.shellInit = ''
-        eval $(${pkgs.thefuck}/bin/thefuck --alias ${cfg.alias})
-      '';
+      environment.shellInit = initScript;
+
+      programs.zsh.shellInit = mkIf prg.zsh.enable initScript;
+      programs.fish.shellInit = mkIf prg.fish.enable initScript;
```

Comment:
Did you test this with fish?
The syntax looks differently here: https://github.com/nvbn/thefuck/wiki/Shell-aliases#fish